### PR TITLE
rustc_plugin: Remove support for syntactic plugins

### PR DESCRIPTION
**Pull request by petrochenkov**
This part of the plugin interface was successfully replaced by token-based procedural macros in theory and in practice.

cc https://github.com/rust-lang/rust/issues/29597
cc https://github.com/rust-lang/rust/pull/64675
r? @Centril

### CODE REVIEW

**Comment by petrochenkov** [comment]
This is a full version of https://github.com/rust-lang/rust/pull/64665 (@matklad @Mark-Simulacrum )

File: src/librustc_plugin_impl/registry.rs [hunk 1]
```diff
@@ -2,10 +2,6 @@
 
 use rustc::lint::LintStore;
 use rustc::session::Session;
-
-use syntax_expand::base::{SyntaxExtension, SyntaxExtensionKind, NamedSyntaxExtension};
```
Comment (Centril) [inline src/librustc_plugin_impl/registry.rs]: The crate should be removable from the `Cargo.toml` now.

File: src/librustc_plugin_impl/registry.rs [hunk 2]
```diff
@@ -33,9 +29,6 @@ pub struct Registry<'a> {
     #[doc(hidden)]
     pub krate_span: Span,
 
-    #[doc(hidden)]
-    pub syntax_exts: Vec<NamedSyntaxExtension>,
-
     #[doc(hidden)]
     pub llvm_passes: Vec<String>,
```
Comment (Centril) [inline src/librustc_plugin_impl/registry.rs]: Aside: is this used for anything? (If not, maybe we should remove it in different PR?)

File: src/test/ui-fulldeps/plugin-args-2.rs [hunk 3]
```diff
@@ -1,10 +1,8 @@
-// run-pass
-// aux-build:plugin-args.rs
+// check-pass
```
Comment (Centril) [inline src/test/ui-fulldeps/plugin-args-2.rs]: File can be deleted; it's identical to `plugin-args-1.rs`.

File: src/test/ui-fulldeps/plugin-as-extern-crate.rs [hunk 4]
```diff
@@ -1,11 +1,11 @@
-// aux-build:attr-plugin-test.rs
+// aux-build:empty-plugin.rs
 // ignore-cross-compile
 //
 // attr_plugin_test will not compile on a cross-compiled target because
```
Comment (Centril) [inline src/test/ui-fulldeps/plugin-as-extern-crate.rs]: Comment needs updating?

**Review by Centril — COMMENTED** [review]
🎉 💖

**Comment by Centril** [comment]
r=me with comments addressed (except for the second re. llvm)

File: src/librustc_plugin_impl/registry.rs (hunk 2 above)
Comment (petrochenkov) [reply src/librustc_plugin_impl/registry.rs]: Yeah, I've seen it, the `krate_span` field above also seems unused.
I'll investigate this and make a separate PR later (next weekend perhaps).

**Comment by petrochenkov** [comment]
ping @rust-lang/compiler @rust-lang/lang
This PR removes user-defined syntax extensions.
Please leave a comment if you think this needs an FCP.

**Comment by joshtriplett** [comment]
I think I remember us already signing off on the removal of plugins in another issue? I don't think we need to sign off again.

File: src/test/ui-fulldeps/plugin-args-2.rs (hunk 3 above)
Comment (petrochenkov) [reply src/test/ui-fulldeps/plugin-args-2.rs]: Well, there are extra argument parentheses in `plugin-args-2`.
I'll remove it together with support for plugin arguments.

**Pull request merged**